js/components/portal-dashboard/student-names.tsx

```diff
@@ -1,12 +1,16 @@
 import React from "react";
 import { getFormattedStudentName } from "../../util/student-utils";
+import { DashboardViewMode } from "../../util/misc";
 
 import css from "../../../css/portal-dashboard/student-names.less";
 
 interface IProps {
   students: any; // TODO: add type
   isAnonymous: boolean;
   isCompact: boolean;
+  setCurrentStudent: (studentId: string) => void;
+  setDashboardViewMode: (mode: DashboardViewMode) => void;
+  setListViewMode: (mode: boolean) => void;
```

Comment:
See my comments below about naming this `toggleQuestionMode` - unless this somehow clashes with upcoming feedback work.

Reply:
How about `showQuestionMode`?

Reply:
It's more like a set or toggle though, isn't it?  We're not showing/hiding question mode.  We're either in question mode or we're in student mode, but in this case we're trying to handle it with a single boolean.  Show/hide sort of implies that a piece of the UI is either there or isn't there.  
Ultimately this probably would be best if it remained `setListViewMode`, but rather than taking a boolean, it took a `ListViewMode` type.  You could model it exactly as I implemented `DashboardViewMode`.  So in misc.ts, add something like:
`export type ListViewMode = "Question" | "Student";`

In app.tsx instead of `inQuestionMode: boolean;` you would have `listViewMode : ListViewMode`.  And then setListViewMode would be something like:
```
  private setListViewMode = (value: ListViewMode) => {
    this.setState({ listViewMode : value });
  }
```

That is really the cleanest way to do it.  Looking now, that doesn't seem to be much more work than changing the name of the function.  So I would to leave `setListViewMode`, but change it so it takes a `ListViewMode` type.  And then have the list view mode stored in state in app.tsx as `listViewMode: ListViewMode`;

Reply:
This would allow us to expand the list view mode in the future if needed and it more clearly describes what is happening.  We have to list view mode types and we are setting one of the other when a button is pressed.  The UI is then updated based on the list view mode type.  Much cleaner, more descriptive, and extensible than using a boolean.